File: Payload_Type/thanatos/mythic/command/cat/cat.go
package cat

import (
	agentstructs "github.com/MythicMeta/MythicContainer/agent_structs"
)

func init() {
	agentstructs.AllPayloadData.Get("thanatos").AddCommand(agentstructs.Command{
		Name:        "cat",
		Description: "Display the contents of a file.",
		HelpString:  "cat [file path]",
		Version:     1,

		MitreAttackMappings: []string{
			"T1005",
			"T1039",
			"T1025",
		},

		Author: "@M_alphaaa",

		CommandAttributes: agentstructs.CommandAttribute{
			SupportedOS: []string{
				agentstructs.SUPPORTED_OS_LINUX, agentstructs.SUPPORTED_OS_WINDOWS,
			},
		},

		CommandParameters: []agentstructs.CommandParameter{
			{
				Name:             "file",
				Description:      "File to display the contents of.",
				ModalDisplayName: "File to display the contents of.",
				ParameterType:    agentstructs.COMMAND_PARAMETER_TYPE_STRING,
				ParameterGroupInformation: []agentstructs.ParameterGroupInfo{
					{
						ParameterIsRequired: true,
						UIModalPosition:     1,
					},
				},
			},
		},

		TaskFunctionParseArgString: func(args *agentstructs.PTTaskMessageArgsData, input string) error {
			args.SetArgValue("file", input)
			return nil
		},

		TaskFunctionParseArgDictionary: func(args *agentstructs.PTTaskMessageArgsData, input map[string]interface{}) error {
			args.LoadArgsFromDictionary(input)
			return nil
		},

		TaskFunctionCreateTasking: func(task *agentstructs.PTTaskMessageAllData) agentstructs.PTTaskCreateTaskingMessageResponse {
			response := agentstructs.PTTaskCreateTaskingMessageResponse{
				Success: true,
				TaskID:  task.Task.ID,
			}

			catFile, _ := task.Args.GetStringArg("file")
			response.DisplayParams = &catFile
			return response
		},
	})
}

File: Payload_Type/thanatos/mythic/thanatos/commands/cat.py
from mythic_container.MythicCommandBase import (
    TaskArguments,
    CommandBase,
    CommandAttributes,
    CommandParameter,
    ParameterType,
    ParameterGroupInfo,
    SupportedOS,
    PTTaskMessageAllData,
    PTTaskCreateTaskingMessageResponse,
)

# TODO: Refactor implementation


class CatArguments(TaskArguments):
    def __init__(self, command_line, **kwargs):
        super().__init__(command_line, **kwargs)
        self.args = [
            CommandParameter(
                name="file",
                type=ParameterType.String,
                description="File to display the contents of.",
                display_name="File to display the contents of.",
                parameter_group_info=[ParameterGroupInfo(ui_position=1)],
            ),
        ]

    async def parse_arguments(self):
        if self.tasking_location in ("command_line", "browserscript"):
            if self.command_line[0] == "{":
                self.load_args_from_dictionary(self.command_line)
            else:
                self.set_arg("file", self.get_command_line())

    async def parse_dictionary(self, dictionary_arguments):
        self.load_args_from_dictionary(dictionary_arguments)


class CatCommand(CommandBase):
    cmd = "cat"
    needs_admin = False
    help_cmd = "cat [file path]"
    description = "Display the contents of a file."
    version = 2
    author = "@M_alphaaa"
    supported_ui_features = ["cat"]
    argument_class = CatArguments
    attackmapping = ["T1005", "T1039", "T1025"]
    attributes = CommandAttributes(
        supported_os=[SupportedOS.Linux, SupportedOS.Windows],
    )

    async def create_go_tasking(
        self, task_data: PTTaskMessageAllData
    ) -> PTTaskCreateTaskingMessageResponse:
        return PTTaskCreateTaskingMessageResponse(
            TaskID=task_data.Task.ID,
            DisplayParams=str(task_data.args.get_arg("file")),
            Success=True,
        )
